enterprise/server/api/api_server.go

```diff
@@ -315,6 +331,56 @@ func (s *APIServer) GetFile(req *apipb.GetFileRequest, server apipb.ApiService_G
 	})
 }
 
+func (s *APIServer) DeleteFile(ctx context.Context, req *apipb.DeleteFileRequest) (*apipb.DeleteFileResponse, error) {
+	ctx, _ = prefix.AttachUserPrefixToContext(ctx, s.env)
+	if _, err := s.checkPreconditions(ctx); err != nil {
+		return nil, err
+	}
+	if err := s.authorizeWrites(ctx); err != nil {
+		return nil, err
+	}
+
+	parsedURL, err := url.Parse(req.GetUri())
+	if err != nil {
+		return nil, status.InvalidArgumentErrorf("Invalid URL")
+	}
+	urlStr := strings.TrimPrefix(parsedURL.RequestURI(), "/")
+
+	var parsedResourceName *digest.ResourceName
+	var cacheType interfaces.CacheType
+	if parsedURL.Scheme == "actioncache" {
+		parsedResourceName, err = digest.ParseActionCacheResourceName(urlStr)
+		if err != nil {
+			return nil, status.InvalidArgumentErrorf("Invalid URL. Does not match expected actioncache URI pattern: %s", err.Error())
+		}
+		cacheType = interfaces.ActionCacheType
+	} else if parsedURL.Scheme == "bytestream" {
+		parsedResourceName, err = digest.ParseDownloadResourceName(urlStr)
+		if err != nil {
+			return nil, status.InvalidArgumentErrorf("Invalid URL. Does not match expected CAS URI pattern: %s", err.Error())
+		}
+		cacheType = interfaces.CASCacheType
+	} else {
```

Comment:
do we need the "schemes" in these urls? if we can distinguish them just by the "/ac/" part that is in the action cache urls, that might be a cleaner API for users to use / learn. Also prevents us from having to worry about them accidentally setting the "host" or anything in that part of the URL.

Reply:
Done